src/popup/MainContainer.tsx

```diff
@@ -57,41 +65,65 @@ export default class MainContainer extends Component<IProps, {}> {
             <Route exact path="/add-token" component={AddToken} />
           </Switch>
         </Router>
+        <UnexpectedErrorDialog store={ this.props.store } />
       </div>
     );
   }
 
   private handleMessage = (request: any) => {
```

Comment:
can we create a store for MainContainer and move this func in there? preferably, routing should happen in the store layer instead of referencing the router from the UI component.